src/transform/src/map_lifting.rs

```diff
@@ -368,7 +489,46 @@ impl LiteralLifting {
                             .scalar_type,
                     ));
                 }
+                if aggregates.is_empty() {
+                    while group_key.last().map(|k| k.is_literal()) == Some(true) {
+                        let key = group_key.pop().unwrap();
+                        result.push(key);
+                    }
+                }
                 result.reverse();
+
+                // Add a Map operator with the remaining literals so that they are lifted in
+                // the next invocation of this transform.
+                let non_literal_keys = group_key.iter().filter(|x| !x.is_literal()).count();
+                if non_literal_keys != group_key.len() {
+                    let first_projected_literal: usize = non_literal_keys + aggregates.len();
+                    let mut projection = Vec::new();
+                    let mut projected_literals = Vec::new();
+                    let mut new_key_pos: usize = 0;
+                    let mut new_group_key = Vec::new();
+                    for key in group_key.iter() {
+                        if key.is_literal() {
+                            projection.push(first_projected_literal + projected_literals.len());
+                            projected_literals.push(key.clone());
+                        } else {
+                            new_group_key.push(key.clone());
+                            projection.push(new_key_pos);
+                            new_key_pos += 1;
+                        }
+                    }
+                    // The new group key without literals
+                    *group_key = new_group_key;
+
```

Comment:
Does this code lift non-terminal literals in `aggregates`?

Reply:
IIUIC, non-terminal literals in `aggregates` are constructions such as `any(<literal>)` or `all(<literal>)`. Those are lifted only if the form a suffix in the list of `aggregates`. I have the code that needs to be added here to permute them, but I'm not able to come up with a test case where that code is exercised. In all the examples I've been able to write, they always appear as a standalone aggreate function in the list of a aggregates:

```
 %5 = Let l3 =                                         +
 | Join %3 %4                                          +
 | | implementation = Unimplemented                    +
 | Map 1                                               +
 | Reduce group=(#0)                                   +
 | | agg any((#0 >= #1))                               +
```

Reply:
Just committed the code that covers that case.